cmd/nerdctl/build_test.go

```diff
@@ -140,6 +140,38 @@ CMD ["echo", "nerdctl-build-test-stdin"]
 	base.Cmd("build", "-t", imageName, "-f", "-", ".").CmdOption(testutil.WithStdin(strings.NewReader(dockerfile))).AssertCombinedOutContains(imageName)
 }
 
+func TestBuildWithDockerfile(t *testing.T) {
+	testutil.RequiresBuild(t)
+	base := testutil.NewBase(t)
+	defer base.Cmd("builder", "prune").Run()
+	imageName := testutil.Identifier(t)
+	defer base.Cmd("rmi", imageName).Run()
+
+	dockerfile := fmt.Sprintf(`FROM %s
+CMD ["echo", "nerdctl-build-test-dockerfile"]
+	`, testutil.CommonImage)
+
+	buildCtx := filepath.Join(t.TempDir(), "test")
+	err := os.MkdirAll(buildCtx, 0755)
+	assert.NilError(t, err)
+	err = os.WriteFile(filepath.Join(buildCtx, "Dockerfile"), []byte(dockerfile), 0644)
+	assert.NilError(t, err)
+
+	pwd, err := os.Getwd()
+	assert.NilError(t, err)
+	err = os.Chdir(buildCtx)
+	assert.NilError(t, err)
+	defer os.Chdir(pwd)
+
+	// hack os.Getwd return "(unreachable)" on rootless
```

Comment:
How does this happen?

Can't find "unreachable" error in the code
https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/os/getwd.go;l=22

Reply:
I actually don't know what happened, `Getwd` returns a strange path during rootless testing.

https://github.com/containerd/nerdctl/runs/7196804919?check_suite_focus=true#step:5:1004